submission/submission-handlers/src/main/kotlin/ac/uk/ebi/biostd/handlers/listeners/SecurityNotificationListener.kt

```diff
@@ -1,25 +1,46 @@
 package ac.uk.ebi.biostd.handlers.listeners
 
+import ac.uk.ebi.biostd.common.events.BIOSTUDIES_EXCHANGE
+import ac.uk.ebi.biostd.handlers.common.HANDLERS_SUBSYSTEM
+import ac.uk.ebi.biostd.handlers.common.SYSTEM_NAME
+import ac.uk.ebi.biostd.handlers.config.NOTIFICATIONS_FAILED_REQUEST_ROUTING_KEY
 import ac.uk.ebi.biostd.handlers.config.SECURITY_NOTIFICATIONS_QUEUE
+import ebi.ac.uk.commons.http.slack.Alert
+import ebi.ac.uk.commons.http.slack.NotificationsSender
 import ebi.ac.uk.extended.events.SecurityNotification
 import ebi.ac.uk.extended.events.SecurityNotificationType.ACTIVATION
 import ebi.ac.uk.extended.events.SecurityNotificationType.ACTIVATION_BY_EMAIL
 import ebi.ac.uk.extended.events.SecurityNotificationType.PASSWORD_RESET
 import ebi.ac.uk.notifications.service.SecurityNotificationService
 import mu.KotlinLogging
 import org.springframework.amqp.rabbit.annotation.RabbitListener
+import org.springframework.amqp.rabbit.core.RabbitTemplate
 
 private val logger = KotlinLogging.logger {}
+private const val ERROR_MESSAGE = "Problem processing security notification of type %s for user %s"
 
-class SecurityNotificationListener(private val securityNotificationService: SecurityNotificationService) {
+class SecurityNotificationListener(
+    private val rabbitTemplate: RabbitTemplate,
+    private val notificationsSender: NotificationsSender,
+    private val securityNotificationService: SecurityNotificationService,
+) {
     @RabbitListener(queues = [SECURITY_NOTIFICATIONS_QUEUE])
     fun receiveMessage(notification: SecurityNotification) {
-        logger.info { "processing ${notification.type} notification for user '${notification.email}'" }
+        logger.info { "Processing ${notification.type} notification for user '${notification.email}'" }
 
-        when (notification.type) {
-            ACTIVATION -> securityNotificationService.sendActivationNotification(notification)
-            ACTIVATION_BY_EMAIL -> securityNotificationService.sendActivationByEmailNotification(notification)
-            PASSWORD_RESET -> securityNotificationService.sendPasswordResetNotification(notification)
-        }
+        runCatching {
```

Comment:
this is not retrying but rather sending a message when there is failure. Are this the expectaions?

Reply:
This is the expected approach. The idea of this ticket is to use the same mechanism that we use for the release notifications to register any failure (which we've historically had very few) in the notifications-failed-log queue instead of keeping retrying indefinitely.

This is actually a very similar approach to the one you suggested where the "parking lot" queue is the failed notifications log, we would just need to add automation to it. I've created [a ticket ](https://www.pivotaltracker.com/story/show/185696865) for it.

Reply:
I was thinking that we were going to actually stop sending message there as no one actually see them but this is the discussed approach that is ok